src/blockencodings.h

```diff
@@ -76,7 +76,7 @@ struct PrefilledTransaction {
     uint16_t index;
     CTransactionRef tx;
 
-    SERIALIZE_METHODS(PrefilledTransaction, obj) { READWRITE(COMPACTSIZE(obj.index), Using<TransactionCompression>(obj.tx)); }
+    SERIALIZE_METHODS(PrefilledTransaction, obj) { READWRITE(COMPACTSIZE(obj.index), TX_WITH_WITNESS(Using<TransactionCompression>(obj.tx))); }
```

Comment:
Not clear -- you could have `BIP_XXX_TX_COMPRESSION` be a `BlockTransactionsSerParams` and have the logic for compression-or-not happen in `BlockTransactions::Serialize`, or it could be a separate formatter. In either case the `TX_WITH_WITNESS` flag could go in with the logic there still, I think.